cluster_traffic_capture/cluster_traffic_capture/gen_haproxy_cfg.py

```diff
@@ -0,0 +1,182 @@
+
+
+def _gen_global_config() -> str:
+    """
+    Generates the HAProxy global/default configuration.
+    """
+
+    return """
+global
+    # Logging configuration
+    log 127.0.0.1:514 local0 # syslogd, facility local0
+    log stdout format raw local0 debug # stdout too
+
+    # Turn on Master/Worker mode.  Required to use SPOE Mirroring.  While hardly an expert, I've attempted to configure
+    # this so that the HAProxy container will exit if a process associated w/ the proxy mechanism runs into trouble.
+    # This will ensure problems are surfaced quickly/obviously, possibly at the expense of resilience at the container
+    # level.
+    #
+    # The premise of master/worker is that it sets up a main process which "is" HAProxy from the perspective of the
+    # rest of the system, and the actual work of HAProxy is done by subprocesses that can be killed/restarted at-will
+    # without changing the process id.  This provides better integration w/ systemd's process management system, and
+    # more logically bundles the various aspects of HAProxy under one process umbrella (the main process).
+    master-worker # not using no-exit-on-failure so container dies if any subprocess dies
+    mworker-max-reloads 3 # Limit subprocess reloads to 3 before SIGTERM is sent
+    hard-stop-after 15s # If a subprocess is still running 15s after reload signal, send SIGTERM
+
+defaults
+    mode http
+    timeout client 10s
+    timeout connect 5s
+    timeout server 10s
+    timeout http-request 10s
+    log global
+
+"""
+
+
+def _get_fe_config_haproxy(haproxy_port: int, mirror: bool) -> str:
+    """
+    Generates the frontend configuration for the HAProxy server (e.g. the rules for it to accept traffic).  If the
+    mirror argument is set to True, the generated configuration will also support mirroring.
+    """
+
+    config = f"""
+# This section outlines the "frontend" for this instance of HAProxy and specifies the rules by which it receives
+# traffic
+frontend haproxy
+    bind :{haproxy_port}
+
+    # Set up the logging for the req/res stream to the primary cluster
+    declare capture request len 80000
+    declare capture response len 80000
+    http-request capture req.body id 0
+    log-format Request-URI:\\ %[capture.req.uri]\\nRequest-Method:\\ %[capture.req.method]\\nRequest-Body:\\ %[capture.req.hdr(0)]\\nResponse-Body:\\ %[capture.res.hdr(0)]
+    
+    # Associate this frontend with the primary cluster
+    default_backend primary_cluster
+
+"""
+
+    mirror_config = """
+    # Required for the mirror SPOA to have access to the request
+    option http-buffer-request
+
+    # Tell HAProxy to also send traffic to this frontend to the mirror agent too
+    filter spoe engine mirror config /usr/local/etc/haproxy/mirror.conf
+
+"""
+
+    return config if not mirror else "\n".join([config, mirror_config])
+
+
+def _gen_be_config_cluster(cluster_nodes: List[HostAddress]) -> str:
+    """
```

Comment:
I agree, this isn't great.  If folks find this particularly unreadable we can move to something like Jinja per your suggestion.

Reply:
I agree w/ Mikayla on docstrings vs template strings.  It may be that the colorification in the IDEs are much more helpful than in the PR review, but it's tough here.

Reply:
@gregschohn I'll create a JIRA to address

Reply:
Task create - https://opensearch.atlassian.net/browse/MIGRATIONS-982